pkg/protocol/engine/consensus/epochgadget/epochorchestrator/performance/performance.go

```diff
@@ -0,0 +1,177 @@
+	}
+
+	committee.ForEach(func(accountID iotago.AccountID, pool *account.Pool) bool {
+		intermediateFactors := make([]uint64, 0)
+		for slot := epochSlotStart; slot <= epochSlotEnd; slot++ {
+			performanceFactorStorage := t.performanceFactorsFunc(slot)
+			if performanceFactorStorage == nil {
+				intermediateFactors = append(intermediateFactors, 0)
+			}
+
+			pf, err := performanceFactorStorage.Load(accountID)
+			if err != nil {
+				panic(errors.Wrapf(err, "failed to load performance factor for account %s", accountID))
+			}
+
+			intermediateFactors = append(intermediateFactors, pf)
+
+		}
+
+		ads.NewMap[iotago.AccountID, PoolRewards](t.rewardsStorage(epoch)).Set(accountID, &PoolRewards{
+			PoolStake:   pool.PoolStake,
+			PoolRewards: t.poolReward(epochSlotEnd, committee.TotalValidatorStake(), committee.TotalStake(), pool.PoolStake, pool.ValidatorStake, pool.FixedCost, t.aggregatePerformanceFactors(intermediateFactors)),
+			FixedCost:   pool.FixedCost,
+		})
+
+		return true
+	})
+}
+
+func (t *Tracker) EligibleValidatorCandidates(_ iotago.EpochIndex) *advancedset.AdvancedSet[iotago.AccountID] {
+	// TODO: we should choose candidates we tracked performance for
+
+	return &advancedset.AdvancedSet[iotago.AccountID]{}
+}
+
+func (t *Tracker) poolStats(epoch iotago.EpochIndex) (poolStats *PoolsStats, err error) {
+	poolStats = new(PoolsStats)
+	poolStatsBytes, err := t.poolStatsStore.Get(epoch.Bytes())
+	if err != nil {
+		return poolStats, errors.Wrapf(err, "failed to get pool stats for epoch %d", epoch)
+	}
+
+	if _, err := poolStats.FromBytes(poolStatsBytes); err != nil {
+		return poolStats, errors.Wrapf(err, "failed to parse pool stats for epoch %d", epoch)
+	}
+
+	return poolStats, nil
+}
+
+func (t *Tracker) LoadCommitteeForEpoch(epoch iotago.EpochIndex) (committee *account.Accounts, exists bool) {
+	accountsBytes, err := t.committeeStore.Get(epoch.Bytes())
+	if err != nil {
+		if errors.Is(err, kvstore.ErrKeyNotFound) {
+			return nil, false
+		}
+		panic(errors.Wrapf(err, "failed to load committee for epoch %d", epoch))
+	}
+
+	committee = account.NewAccounts()
+	if _, err = committee.FromBytes(accountsBytes); err != nil {
+		panic(errors.Wrapf(err, "failed to parse committee for epoch %d", epoch))
+	}
+
+	return committee, true
+}
+
+func (t *Tracker) storeCommitteeForEpoch(epochIndex iotago.EpochIndex, committee *account.Accounts) error {
+	committeeBytes, err := committee.Bytes()
+	if err != nil {
+		return err
+	}
+
+	return t.committeeStore.Set(epochIndex.Bytes(), committeeBytes)
+}
+
+func (t *Tracker) aggregatePerformanceFactors(issuedBlocksPerSlot []uint64) uint64 {
+	var sum uint64
+	for _, issuedBlocks := range issuedBlocksPerSlot {
+		if issuedBlocks > uint64(validatorBlocksPerSlot) {
+			// we harshly punish validators that issue any blocks more than allowed
+			return 0
+		}
+		sum += issuedBlocks
+	}
+
+	return sum / uint64(len(issuedBlocksPerSlot))
```

Comment:
Even if it may never happen, but should we add a check at the beginning of the function to avoid division by zero?

```golang
if (len(issuedBlocksPerSlot)==0) {
    return 0
}
```